components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -742,6 +744,12 @@ class GeckoEngineSession(
         return cookiesPolicies
     }
 
+    internal fun GeckoSession.ProgressDelegate.SecurityInformation.getIssuerName(): String? {
+        return certificate?.issuerDN?.name?.substringAfterLast("O=")?.substringBeforeLast(",C=")
+        // I can't get the below implementation to work correctly
+        // certificate?.issuerDN?.name?.let { SslCertificate.DName(it).getDName() }
```

Comment:
This seems to work using:
```
return SslCertificate(certificate).issuedBy.oName
```

You will need to create an instance of `SslCertificate` first to create an instance of `DName`. Tested and this works in the app, but we need to figure out how to mock this correctly for unit testing.

Reply:
At least with the androidx test runner I can't get a unit test running that depends on `android.net.http.SslCertificate.DName`. I am getting `java.lang.NoClassDefFoundError: com/android/org/bouncycastle/asn1/x509/X509Name`... tried for a while to resolve this without much luck.

So, I am OK landing this simple parser for now which we can cover with tests at least. @pocmo wdyt?

Reply:
With the simple parser I am not sure whether we will be guaranteed to always get the string ordered in the same way with all elements always present. If not that could be causing issues. If we want to land this then maybe we should get another review from the GV team to make sure this is okay.

The other option here is to move this specific test to `androidTest` to run it on an actual device/emulator where those classes exist. We already run other `androidTest` tests for `browser-engine-gekco-nightly` on Firebase (and others hopefully soon too). I think I would prefer that. What do you think?

Reply:
@pocmo and I talked about this: Let's land the PR as is with the simple solution for now. 

Otherwise we have to parse the full SSL certificate every time, which is costly. We can address this in https://github.com/mozilla-mobile/android-components/issues/5557 and also talk to GV team again to see if they can provide a better API for us.

@kglazko There are some lint issues to address and would you mind squashing your commits then we can land this :).

Reply:
Thank you! Sounds great.